### PR TITLE
Show distinctive icons on dashboard for different builder types

### PR DESCRIPTION
**Pull request by yusufm**
Test=flutter test
Bug:https://github.com/flutter/flutter/issues/81062

image

### CODE REVIEW

File: app_flutter/lib/widgets/task_icon.dart [hunk 1]
```diff
@@ -44,22 +44,19 @@ class TaskIcon extends StatelessWidget {
...
-      case StageName.devicelabWin:
-        return Image.asset(
-          'assets/windows.png',
-        );
-      case StageName.devicelabIOs:
-        return Image.asset(
-          'assets/apple.png',
-          color: blendFilter,
-        );
+        if (qualifiedTask.builder.toLowerCase().startsWith('linux')) {
```
Comment (CaseyHillers) [inline app_flutter/lib/widgets/task_icon.dart:47]: We no longer need the switch for `StageName`, let's remove it

File: app_flutter/lib/widgets/task_icon.dart [hunk 2]
```diff
@@ -44,22 +44,19 @@ class TaskIcon extends StatelessWidget {
...
+            'assets/linux.png',
+          );
+        } else if (qualifiedTask.builder.toLowerCase().startsWith('mac')) {
+          return Image.asset(
+            'assets/apple.png',
+          );
+        } else if (qualifiedTask.builder.toLowerCase().startsWith('win')) {
+          return Image.asset(
+            'assets/windows.png',
+          );
```
Comment (CaseyHillers) [inline app_flutter/lib/widgets/task_icon.dart:58]: Can you fix Fuchsia and add the web tests?

File: app_flutter/lib/widgets/task_icon.dart [hunk 3]
```diff
@@ -44,22 +44,19 @@ class TaskIcon extends StatelessWidget {
           color: blendFilter,
         );
       case StageName.luci:
-        return Image.asset(
-          'assets/chromium.png',
-        );
-      case StageName.devicelab:
-        return Image.asset(
-          'assets/android.png',
```
Comment (CaseyHillers) [inline app_flutter/lib/widgets/task_icon.dart:52]: Is there a builder pattern to show android tests?

File: app_flutter/lib/widgets/task_icon.dart [hunk 4]
```diff
@@ -44,22 +44,19 @@ class TaskIcon extends StatelessWidget {
...
-          'assets/apple.png',
-          color: blendFilter,
-        );
+        if (qualifiedTask.builder.toLowerCase().startsWith('linux')) {
+          return Image.asset(
+            'assets/linux.png',
+          );
+        } else if (qualifiedTask.builder.toLowerCase().startsWith('mac')) {
+          return Image.asset(
+            'assets/apple.png',
```
Comment (CaseyHillers) [inline app_flutter/lib/widgets/task_icon.dart:53]: nit: `contains('_ios')` we could show an iPhone

File: app_flutter/lib/widgets/task_icon.dart (hunk 1 above)
Comment (yusufm) [reply app_flutter/lib/widgets/task_icon.dart:47]: done

File: app_flutter/lib/widgets/task_icon.dart (hunk 2 above)
Comment (yusufm) [reply app_flutter/lib/widgets/task_icon.dart:58]: Not sure what you mean by fix fuchsia?

File: app_flutter/lib/widgets/task_icon.dart (hunk 3 above)
Comment (yusufm) [reply app_flutter/lib/widgets/task_icon.dart:52]: There are android on windows, linux, etc. So not sure if having an android icon for all those platforms would be helpful. Ideally we would want a distinctive win+android, win+linux, etc. icon for those cases.

File: app_flutter/lib/widgets/task_icon.dart (hunk 2 above)
Comment (yusufm) [reply app_flutter/lib/widgets/task_icon.dart:58]: I simplified the options, ptal.

File: app_flutter/lib/widgets/task_icon.dart (hunk 3 above)
Comment (CaseyHillers) [reply app_flutter/lib/widgets/task_icon.dart:52]: There's better signal when differentiating by device than host (we can reuse the android logo for mac-android, windows-android, and linux-android)

File: app_flutter/lib/widgets/task_icon.dart (hunk 1 above)
Comment (CaseyHillers) [reply app_flutter/lib/widgets/task_icon.dart:47]: The switch is still here

File: app_flutter/lib/widgets/task_icon.dart [hunk 5]
```diff
@@ -44,22 +44,19 @@ class TaskIcon extends StatelessWidget {
...
-          'assets/windows.png',
-        );
-      case StageName.devicelabIOs:
-        return Image.asset(
-          'assets/apple.png',
-          color: blendFilter,
-        );
+        if (qualifiedTask.builder.toLowerCase().startsWith('linux')) {
+          return Image.asset(
+            'assets/linux.png',
```
Comment (CaseyHillers) [inline app_flutter/lib/widgets/task_icon.dart:49]: Can you ensure all images have `color: blendFilter` to ensure dark mode support

File: app_flutter/lib/widgets/task_icon.dart (hunk 3 above)
Comment (yusufm) [reply app_flutter/lib/widgets/task_icon.dart:52]: done

File: app_flutter/lib/widgets/task_icon.dart (hunk 5 above)
Comment (yusufm) [reply app_flutter/lib/widgets/task_icon.dart:49]: done

File: app_flutter/lib/widgets/task_icon.dart (hunk 2 above)
Comment (CaseyHillers) [reply app_flutter/lib/widgets/task_icon.dart:56]: Fuchsia icon should be set for `contains('_fuchsia')`. Previous filter was old based on legacy luci builders

Comment (yusufm) [reply app_flutter/lib/widgets/task_icon.dart:61]: done

File: app_flutter/lib/widgets/task_icon.dart [hunk 6]
```diff
@@ -32,35 +32,42 @@ class TaskIcon extends StatelessWidget {
...
+    if (qualifiedTask.builder != null) {
+      if (qualifiedTask.builder.toLowerCase().contains('_fuchsia')) {
         return Image.asset(
-          'assets/cirrus.png',
+          'assets/fuchsia.png',
           color: blendFilter,
         );
-      case StageName.luci:
-        return Image.asset(
-          'assets/chromium.png',
```
Comment (CaseyHillers) [inline app_flutter/lib/widgets/task_icon.dart:48]: The web tests are missing. `contains('_web')` should be sufficient

Comment (yusufm) [reply app_flutter/lib/widgets/task_icon.dart:48]: done

**Pull request merged**